public/js/controllers/data_controller.js

```diff
@@ -146,7 +147,21 @@ export class DataController {
             }, error => console.log(error));
     }
 
+    static watchProjectsWithATconfig(activeProjects) {
+        let projectsWithATconfig = new Set();
+        activeProjects.forEach((project) => {
+            if (project.hasOwnProperty("africas_talking")) {
+                projectsWithATconfig.add(project.project_name);
+            }
+        });
+        DataController.projectsWithATconfig = [...projectsWithATconfig];
```

Comment:
I think this approach is slightly more complicated than it need be, and risks being error prone if you ever end up forgetting to keep this `projectsWithATconfig` property up to date.
  Instead of loading the active_projects, synchronising them with a list of names that match a condition, passing the project_name around, then checking that list, why not just pass the full project object around? Then in the watchATCredits function you can just say `if (project.hasOwnProperty("africas_talking"))...`.

Reply:
Updated, thanks. I've not been able to get the project object but I've resolved the issue. I'll work on using browser storage to store some of the info like active projects objects, this will help to reduce the number of reads we're making.